File: backend/app/schemas/interview.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class InterviewStartRequest(BaseModel):
    scope: str = Field("all", pattern="^(all|weak_points)$")
    num_questions: int = Field(5, ge=1, le=20)


class AnswerSubmitRequest(BaseModel):
    answer: str = Field(..., min_length=1)


class InterviewQuestionOut(BaseModel):
    id: uuid.UUID
    question: str
    user_answer: str | None = None
    ai_score: int | None = None
    ai_feedback: str | None = None
    question_order: int

    model_config = {"from_attributes": True}


class InterviewSessionOut(BaseModel):
    id: uuid.UUID
    title: str
    scope: str
    total_score: int | None = None
    summary: str | None = None
    status: str
    created_at: datetime
    finished_at: datetime | None = None
    questions: list[InterviewQuestionOut] = Field(default_factory=list)

    model_config = {"from_attributes": True}


class WeakPointOut(BaseModel):
    note_id: uuid.UUID
    title: str
    avg_score: float
    times_tested: int

File: frontend/src/api/interview.ts
import client from './client'
import type { ApiResponse } from './notes'

export interface InterviewQuestion {
  id: string
  question: string
  user_answer: string | null
  ai_score: number | null
  ai_feedback: string | null
  question_order: number
}

export interface InterviewSession {
  id: string
  title: string
  scope: 'all' | 'weak_points'
  total_score: number | null
  summary: string | null
  status: 'in_progress' | 'completed'
  created_at: string
  finished_at: string | null
  questions: InterviewQuestion[]
}

export interface WeakPoint {
  note_id: string
  title: string
  avg_score: number
  times_tested: number
}

export const interviewApi = {
  start: (data: { scope: 'all' | 'weak_points'; num_questions: number }) =>
    client.post<any, ApiResponse<InterviewSession>>('/interview/start', data),

  submitAnswer: (sessionId: string, questionId: string, answer: string) =>
    client.post<any, ApiResponse<InterviewQuestion>>(`/interview/${sessionId}/answer/${questionId}`, { answer }),

  finish: (sessionId: string) =>
    client.post<any, ApiResponse<InterviewSession>>(`/interview/${sessionId}/finish`),

  list: () =>
    client.get<any, ApiResponse<InterviewSession[]>>('/interview/sessions'),

  get: (sessionId: string) =>
    client.get<any, ApiResponse<InterviewSession>>(`/interview/sessions/${sessionId}`),

  weakPoints: () =>
    client.get<any, ApiResponse<WeakPoint[]>>('/interview/weak-points'),
}
